### ISSUE TITLE
proposal: `no_leading_underscore_for_local_functions`

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#3250]
## Description

Do not name local functions with a leading underscore: their visibility is dictated by local scope. Similar to `no_leading_underscore_for_local_variables`.

## Details

`no_leading_underscore_for_local_variables` does not trigger on local functions. Rather than treat that as a false negative, which is hard to land, we can just create a new rule.

I often catch this in readability reviews.

## Kind

*Does this enforce style advice?  Guard against errors?  Other?*

Style, I suppose. Meets with Effective Dart recommendations.

## Good Examples 

```dart
void foo10() {
  void foo11() {}
}
```

## Bad Examples

```dart
void foo10() {
  void _foo11() {}
}
```

### Discussion checklist

- [x] List any existing rules this proposal complements, overlaps or conflict with.
- [ ] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [ ] If there's any prior art (e.g., in other linters), please add references here.
- [x] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn’t any corresponding advice, should there be?)
- [x] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

[Writing Lints]: https://github.com/dart-lang/linter/blob/master/doc/WritingLints.MD
[Effective Dart]: https://dart.dev/guides/language/effective-dart
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo]
[SDK Tracker]: https://github.com/dart-lang/sdk/issues

**Comment by srawlins** [comment]
CC @goderbauer I found this while implementing a fix for `no_leading_underscore_for_local_identifiers`.

**Comment by goderbauer** [comment]
I naively thought that `no_leading_underscore_for_local_variables` would already cover it, but since it doesn't I am in factor of this new lint. 

When the lint exists, we should also consider it for inclusion into `package:lints`.

**Comment by srawlins** [comment]
Handled in `no_leading_underscores_for_local_identifiers` in https://github.com/dart-lang/linter/issues/3360
